src/java.management.rmi/share/classes/javax/management/remote/rmi/RMIConnectionImpl.java

```diff
@@ -971,13 +971,17 @@ public Integer[] addNotificationListeners(ObjectName[] names,
         if (names == null || filters == null) {
             throw new IllegalArgumentException("Got null arguments.");
         }
+        // Accept an array of delegationSubjects from e.g. earlier JDKs,
+        // but throw if it contains any non-null values.
         if (delegationSubjects != null) {
-            throw new UnsupportedOperationException("Subject Delegation has been removed.");
+            for (Subject s: delegationSubjects) {
+                if (s != null) {
+                    throw new UnsupportedOperationException("Subject Delegation has been removed.");
+                }
+            }
         }
-        Subject[] sbjs = new Subject[names.length];
-        if (names.length != filters.length || filters.length != sbjs.length) {
-            final String msg =
-                "The value lengths of 3 parameters are not same.";
+        if (names.length != filters.length) {
+            final String msg = "The lengths of names and filters parameters are not same.";
```

Comment:
I wonder if we should check that if present, `delegationSubjects.length == names.length`?

Reply:
That seems pretty extreme -- it's an array we explicitly don't want and are going to ignore?  If somebody passes a non-null member, we will throw as unsupported.  I was thinking that was enough, hence removing the sbjs array to be quite certain we can't pass on any supplied Subjects.

Reply:
Well my thinking was this: the fact that the jconsole tab was blank shows that the array may being passed. The previous code verified that all three arrays had the same length - so it would have failed if the array had a length different than the other two. So I would prefer if we kept on throwing in that case. In other case, we now allow and prefer `null` - but if non-null - we will allow a null-filled array passed by an older client but  we should not accept something that would have been invalid then.

Reply:
That the JConsole tab was blank shows that the older RMIConnector's addListenerWithSubject creates a new single-entry array from the given delegationSubject (which is null) and passes it onwards.  The app is not creating the array itself., that's us.

(also, maybe that JConsole relies on listeners in order to show a screen that doesn't really need to depend on them, but this change is obviously about being compatible with that)

We all know that that is the only use case out there, the current wisdom is that this feature is not used, nobody is creating the Subject array and calling addListenersWithSubjects (plural) with it...

IF we find such an app app, we are going to ignore the array unless it contains a non-null entry.  This seems safe and efficient.  We are documenting that it should be null,  and it is weird to document a length requirement for something that should be null...  8-)

Reply:
This shows that when SubjectDelegation was not used, a null-filled array of the same length than the two other arrays was expected before (in previous versions of the JDK where SubjectDelegation was supported, but in the case where it wasn't used). 
I am not suggesting to document the length requirement. The length requirement was enforced before and undocumented. I'm just suggesting that we allow null and null-filled but don't allow something (null filled array of wrong length) that would have been rejeceted in previous JDK versions.

Reply:
Yes, completely understand.  I just don't think it has any benefit.

Reply:
Hmmm... the spec still says:
```
     * @throws IllegalArgumentException if <code>names</code> or
     * <code>filters</code> is null, or if <code>names</code> contains
     * a null element, or if the three arrays do not all have the same
     * size.
 ```

Reply:
Thanks for additionally spotting the throws IAE needs updating, adding that now... (Will update the CSR also.)